fs/osfs.py

```diff
@@ -368,7 +374,10 @@ def opendir(self, path, factory=None):
 
     def getsyspath(self, path):
         # type: (Text) -> Text
-        sys_path = self._to_sys_path(path)
+        sys_path = os.path.join(
```

Comment:
This method still returns unicode, so no change in behaviour.

If you want to call an `os` method on python2.7 and support broken unicode you will need to process the returned unicode path with `fsencode`. On Python3 fsencode is not required (but won't break).

That would be a source of annoyance. So I'm considering a `getospath` method which `fsencode`s the syspath.

@pombredanne @althonos  Thoughts?